lib/urldata.h

```diff
@@ -201,6 +201,8 @@
 /* Download buffer size, keep it fairly big for speed reasons */
 #undef BUFSIZE
 #define BUFSIZE CURL_MAX_WRITE_SIZE
+#define MAX_BUFSIZE 524288
```

Comment:
Do you have any thoughts on this specific limit? Why 512KB? For SFTP we've seen that even larger buffers help, mostly since we don't do any buffer sliding in the code so libcurl has to drain the entire buffer first before it can refill it.

Also, the max limit is not documented.